solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -798,6 +798,12 @@ private void reloadSecurityProperties() {
     SecurityConfHandler.SecurityConfig securityConfig = securityConfHandler.getSecurityConfig(false);
     initializeAuthorizationPlugin((Map<String, Object>) securityConfig.getData().get("authorization"));
     initializeAuthenticationPlugin((Map<String, Object>) securityConfig.getData().get("authentication"));
+    if (authenticationPlugin != null && authenticationPlugin.plugin.getMetricRegistry() == null) {
```

Comment:
Is this second check necessary? we know that just after the plugin was created its metricRegistry is null, it's set only after `initializeMetrics` has been called.

Reply:
Yea, I think you're right.